TrustModel.md

```diff
@@ -0,0 +1,426 @@
+and assessment or certification by industry or standards bodies[1: 5.1 Goal]. Theoretically, the
+trust policy can be fine-grained to the component level, so there are many combinations of the
+trust policies that a specific solution may adopt basing on component level. In fact, different
+forms of confidential container and specific solutions adopt the trust policy based on the
+granularity of execution stage, that is, all components involved in each stage will follow the
+same trust policy.
+
+
+In practice, those deploying workloads into TEE environments may have varying
+levels of trust in the workload orchestration or cloud personas 
+hosting the
+workload. This trust may be based on factors such as the relationship with the owner or operator of
+the host, the software and hardware it comprises, and the likelihood of physical, software, or
+social engineering compromise[1: 5.1 Goal].
+
+Therefore, the level of trust ultimately determines what kind of trust model to adopt. The trust
+model needs to define their own trust policies for the execution stages. There are many approaches
+to implement trust policies. The most important thing is to ensure the chain of trust is never
+broken, and the validity of TEE in attestation procedure is proved sufficiently. 
+
+
+
+----
+
+## Trusted Execution Environment Boundary
+Core to confidential containers is the use of a Trusted Execution Environment which protects 
+memory inside the environment from access in the clear by any actor outside the environment. 
+ANd core to the trust model is where within a cloud native 
+stack to 
+put the Trusted Execution 
+boundary when looking to deliver confidential containers.
+
+Three main boundaries for the Trusted Execution Boundary are:
+- Node
+- Pod
+- Container/Container Processes
+
+<*Add Diagram*>
+
+
+Each boundary changes requirements on what components must exist inside the TEE, the 
+form the image to run inside the TEE takes and ultimately on the trust model.
+
+### Node boundary
+This uses a virtual machine with hardware TEE capability to run a K8s node.
+The kubelet is inside the TEE boundary and therefore communication between api-server and 
+kubelet would cross the boundary.
+The 
+
+### Pod Boundary
+This uses a micro virtual machine (microVM) with hardware TEE capability.
+The microVM contains an agent 
+
+### Container/Container Process boundary
+This uses an enclave to run a LibOS (Intel SGX Enclave)
+
+
+In all cases the application code interacts with data within a hardware TEE
+
+A brief comparison:
+
+| | node | pod  | container/process |
+| | vm-cc | microvm-cc/kata-cc\*1 | enclave-cc |
+| :-: | :-: | :-: | :-: |
+| TCB footprint | Large | Medium | Small |
+| Location of trust boundary\*2 | between K8s API server and kubelet | agent API\*3 | ECALL and OCALL |
+| Threat Surface | High | Medium | Low |
+| Deployment complexity | High | Low | Medium |
+| Changes to workload Orchestration Experience | Low | Medium | High |
```

Comment:
I'd measure Process as medium. It's just one image build tool (e.g. GSC). It seems like the same level of burden for pod, because you need to change the image build to accommodate the key management (see 3rd party signing in comments above).